---
title: "R Figure 1"
output:
  html_document:
    df_print: paged
  github_document: default
---

```{r libraries}
suppressPackageStartupMessages({
  library(Seurat)
  library(Signac)
  library(BSgenome.Drerio.UCSC.danRer11)
  library(EnhancedVolcano)
  library(ggsci)
  library(patchwork)
})
```

# Read data
```{r read_HB13}
HB13 <- readRDS(file = "../data/HB13hpf_neural.RDS")
DefaultAssay(HB13) <- "SCT"
Idents(HB13) <- "Clusters"
```

```{r read_HB16}
HB16 <- readRDS(file = "../data/HB16hpf_neural.RDS")
DefaultAssay(HB16) <- "SCT"
Idents(HB16) <- "Clusters"
```

# Cluster names

## HB13hpf

Named clusters after clustering at resolution 8, needed to resolve all rhombomeres into separate clusters.
```{r umap_HB13}
DimPlot(HB13, reduction = "wnn.umap") + scale_color_igv()
```
Combining multiple clusters of same cell type.
```{r rename_HB13}
Idents(HB13) <- "Clusters"
HB13 <- RenameIdents(HB13,
                       "CHB.1" = "CHB",
                       "CHB.2" = "CHB",
                       "CHB.3" = "CHB",
                       "SC.1" = "SC",
                       "SC.2" = "SC",
                       "SC.3" = "SC",
                       "r5.1" = "r5",
                       "r5.2" = "r5",
                       "MB.1" = "MB",
                       "MB.2" = "MB",
                       "MB.3" = "MB",
                       "MHB.1" = "MHB",
                       "MHB.2" = "MHB",
                       "MHB.3" = "MHB",
                       "MHB.4" = "MHB",
                       "MHB.5" = "MHB",
                       "FB.1" = "FB",
                       "FB.2" = "FB",
                       "FB.3" = "FB",
                       "FB.4" = "FB")
levels(HB13) <- c("FB","MB","MHB","r1","r1 & r2","r2","r3","r4","r5","r6",
                    "low_expression","unknown","Neuron","Ciliated","CHB","SC")
umap.HB13 <- DimPlot(HB13, reduction = "wnn.umap") + scale_color_igv()
umap.HB13
```

## HB16hpf
Named clusters after clustering at resolution 6, needed to resolve all rhombomeres into separate clusters.
```{r umap_HB16}
DimPlot(HB16, reduction = "wnn.umap") + scale_color_igv()
```
Combining multiple clusters of same cell type.
```{r rename_HB16}
Idents(HB16) <- "Clusters"
HB16 <- RenameIdents(HB16,
                       "CHB.1" = "CHB",
                       "CHB.2" = "CHB",
                       "CHB.3" = "CHB",
                       "CHB.4" = "CHB",
                       "SC.1" = "SC",
                       "SC.2" = "SC",
                       "SC.3" = "SC",
                       "r5.1" = "r5",
                       "r5.2" = "r5",
                       "MB.1" = "MB",
                       "MB.2" = "MB",
                       "MHB.1" = "MHB",
                       "MHB.2" = "MHB",
                       "MHB.3" = "MHB")
levels(HB16) <- c("FB","MB","MHB","r1","r2","r3","r4","r5","r6",
                    "DorsNT & NC","Neuron","Ciliated","CHB","SC")
umap.HB16 <- DimPlot(HB16, reduction = "wnn.umap") + scale_color_igv()
umap.HB16
```
# Gene expression and chromvar activity plots

```{r function}
GetChromVarPlots <- function(object, motifs, reduction){
  DefaultAssay(object) <- "chromvar"
  mylist <- list()
  for(motif in 1:length(motifs)){
    p <- FeaturePlot(
      object = object,
      reduction = reduction,
      features = motifs[[motif]],
      min.cutoff = 'q10',
      max.cutoff = 2,
      pt.size = 1)  +
      ggtitle(paste(motifs[[motif]],names(motifs[motif]), sep = " ")) +
      theme(plot.title = element_text(size = 15)) + NoLegend()
    mylist[[motifs[[motif]]]] <- p
  }
  
  return(mylist)
}
```

```{r function2}
GetChromVarPlotsPlusLegend <- function(object, motifs, reduction){
  DefaultAssay(object) <- "chromvar"
  mylist <- list()
  for(motif in 1:length(motifs)){
    p <- FeaturePlot(
      object = object,
      reduction = reduction,
      features = motifs[[motif]],
      min.cutoff = 'q10',
      max.cutoff = 2,
      pt.size = 1)  +
      ggtitle(paste(motifs[[motif]],names(motifs[motif]), sep = " ")) +
      theme(plot.title = element_text(size = 15))
    mylist[[motifs[[motif]]]] <- p
  }
  
  return(mylist)
}
```

```{r HB13_runchromvar2}
# HB13 <- RunChromVAR(
#   object = HB13,
#   genome = BSgenome.Drerio.UCSC.danRer11,
#   assay = "peaks"
# )
# saveRDS(HB13, file = "../data/HB13hpf_neural.RDS")
```

```{r HB16_runchromvar2}
# HB16 <- RunChromVAR(
#   object = HB16,
#   genome = BSgenome.Drerio.UCSC.danRer11,
#   assay = "peaks"
# )
# saveRDS(HB16, file = "../data/HB16hpf_neural.RDS")
```

## HB13hpf
```{r}
motif1.names <- c("CDX4","HOXB9","RARA::RXRG","HNF1B","EGR2","Mafb")
motifs1 <- c("MA1473.1","MA1503.1","MA1149.1","MA0153.2","MA0472.2","MA0117.2")
names(motifs1) <- motif1.names
```

```{r HB13_feature}
HB13.cdx4 <- FeaturePlot(HB13, features = "cdx4", reduction = "wnn.umap", max.cutoff = 1.3)
HB13.hoxb7a <- FeaturePlot(HB13, features = "hoxb7a", reduction = "wnn.umap", max.cutoff = 1.3)
HB13.hoxb9a <- FeaturePlot(HB13, features = "hoxb9a", reduction = "wnn.umap", max.cutoff = 1.3)
```

```{r HB13_exp, fig.width=10, fig.height=4}
HB13.exp <- HB13.cdx4 + HB13.hoxb7a + HB13.hoxb9a + plot_layout(ncol = 3, guides = "collect")
HB13.exp
```

```{r getchromvarplots, fig.width=10, fig.height=10}
HB13.cv.motif.legend <- GetChromVarPlotsPlusLegend(HB13, motifs1, "wnn.umap")
HB13.cv.list <- GetChromVarPlots(HB13, motifs1, "wnn.umap")
wrap_plots(HB13.cv.motif.legend)
```
```{r HB13_motif1, fig.width=7, fig.height=4}
HB13.motif1 <- HB13.cv.motif.legend[[1]] + HB13.cv.list[[2]] + plot_layout(ncol = 2, guides = "collect")
HB13.motif1
```

```{r HB13_motif2, fig.width=15, fig.height=4}
HB13.motif2 <- HB13.cv.motif.legend[[3]] + HB13.cv.list[[4]] + HB13.cv.list[[5]] + HB13.cv.list[[6]] + 
  plot_layout(ncol = 4, guides = "collect")
HB13.motif2
```

```{r HB13_feature2}
HB13.crabp2a <- FeaturePlot(HB13, features = "crabp2a", reduction = "wnn.umap", max.cutoff = 1.3)
HB13.rorcb <- FeaturePlot(HB13, features = "rorcb", reduction = "wnn.umap", max.cutoff = 1.3)
HB13.egr2b <- FeaturePlot(HB13, features = "egr2b", reduction = "wnn.umap", max.cutoff = 1.3)
HB13.mafba <- FeaturePlot(HB13, features = "mafba", reduction = "wnn.umap", max.cutoff = 1.3)
```

```{r HB13_exp2, fig.width=15, fig.height=4}
HB13.exp2 <- HB13.crabp2a + HB13.rorcb + HB13.egr2b + HB13.mafba + plot_layout(ncol = 4, guides = "collect")
HB13.exp2
```

## HB16hpf

```{r HB16.getchromvarplots, fig.width=10, fig.height=7}
HB16.cv.motif.legend <- GetChromVarPlotsPlusLegend(HB16, motifs1, "wnn.umap")
HB16.cv.list <- GetChromVarPlots(HB16, motifs1, "wnn.umap")
wrap_plots(HB16.cv.motif.legend)
```
```{r HB16_motif, fig.width=7, fig.height=4}
HB16.motif <- HB16.cv.motif.legend[[1]] + HB16.cv.list[[2]]  + 
  plot_layout(ncol = 2, guides = "collect")
HB16.motif
```
```{r HB16_feature}
HB16.crabp2a <- FeaturePlot(HB16, features = "cdx4", reduction = "wnn.umap", max.cutoff = 1.3)
HB16.rorcb <- FeaturePlot(HB16, features = "hoxb9a", reduction = "wnn.umap", max.cutoff = 1.3)
HB16.egr2b <- FeaturePlot(HB16, features = "crabp2a", reduction = "wnn.umap", max.cutoff = 1.3)
HB16.mafba <- FeaturePlot(HB16, features = "rorcb", reduction = "wnn.umap", max.cutoff = 1.3)
```

```{r HB16_exp1, fig.width=7, fig.height=4}
HB16.exp <- HB16.crabp2a + HB16.rorcb + plot_layout(ncol = 2, guides = "collect")
HB16.exp
```

```{r HB16_exp2, fig.width=7, fig.height=4}
HB16.exp2 <- HB16.egr2b + HB16.mafba + plot_layout(ncol = 2, guides = "collect")
HB16.exp2
```
# Figure 1 combined plots
```{r combined, fig.width=10, fig.height=20}
layout <- "
AABBB
AACCC
DDDDD
EEEEE
FFGGG
HHIII"

combined1 <- umap.HB13 + HB13.exp + HB13.motif1 + HB13.exp2 + HB13.motif2 + 
  umap.HB16 + HB16.exp + HB16.motif + HB16.exp2 + 
  plot_layout(design = layout)
combined1
```

```{r combined2, fig.width=15, fig.height=20}
layout <- "
AAAAAABBBB
CCCCCCDDDD
EEEEEEEEEE
FFFFFFFFFF
GGGGGGHHHH
IIIIIIHHHH"

combined2 <- umap.HB13 + umap.HB16 + 
  HB13.exp + HB13.motif1 + 
  HB13.exp2 + 
  HB13.motif2 + 
  HB16.exp + (HB16.motif + plot_layout(ncol = 1, guides = "collect")) + 
  HB16.exp2 + 
  plot_layout(design = layout, heights = c(2,1,1,1,1,1)) +
  theme(title = element_text(size = 3))
combined2
ggsave(filename = "Plots/Figure1.png", width = 15, height = 20, plot = combined2)
```

```{r sessionInfo}
sessionInfo()
```

